.github/CHANGELOG.md

```diff
@@ -2,6 +2,48 @@
 
 <h3>New features since last release</h3>
 
+* The `specs` QNode transform creates a function that produces the specifications for a circuit
+  at given arguments and keywords. The QNode property `specs` can also provide this information
+  after QNode execution.  The tape property `specs` provides a subset of the same quantities.
+  The tape functions `get_resources` and `get_depth` are supereded by `specs` and will be
+  deprecated after one release cycle.
+  [(#1245)](https://github.com/PennyLaneAI/pennylane/pull/1245)
+
+  For example:
+
+  ```python
+  dev = qml.device('default.qubit', wires=4)
+
+  @qml.qnode(dev, diff_method='parameter-shift')
+  def circuit(x, y):
+      qml.RX(x[0], wires=0)
+      qml.Toffoli(wires=(0, 1, 2))
+      qml.CRY(x[1], wires=(0, 1))
+      qml.Rot(x[2], x[3], y, wires=2)
+      return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+  x = np.array([0.05, 0.1, 0.2, 0.3], requires_grad=True)
+  y = np.array(0.4, requires_grad=False)
+
+  specs_func = qml.specs(circuit)
+  info = specs_func(x, y)
+  ```
+
+  ```pycon
+  >>> info
+  {'by_size': defaultdict(int, {1: 2, 3: 1, 2: 1}),
+ 'by_name': defaultdict(int, {'RX': 1, 'Toffoli': 1, 'CRY': 1, 'Rot': 1}),
+ 'total_operations': 4,
+ 'total_observables': 2,
+ 'num_tape_wires': 3,
+ 'depth': 3,
+ 'num_trainable_params': 4,
```

Comment:
Should this be `num_gate_params`? Since, if you use a template like `StronglyEntanglingLayers`, the number of params is not 4 but instead larger, e.g.,
```
 'depth': 21,
 'num_trainable_params': 39,
 'num_parameter_shift_executions': 81,
```

Reply:
This disappears when `diff_method="backprop"` because `tape.trainable_params` does not actually reflect which parameters are trainable.  I'd like to have "num_trainable_params" for backprop as well, but that will take a little more work.